From d67a55cd917189c443142e0ddfbb7ef36109adc9 Mon Sep 17 00:00:00 2001
From: Chen Minqiang <ptpt52@gmail.com>
Date: Sun, 24 Oct 2021 19:49:18 +0800
Subject: [PATCH] ath79: lzma-loader: use -O2 instead of -Os

gcc 11 build with -Os generate broken okli-loader
notice that no issue with gcc 10
here disable -Os option as a workaround

Signed-off-by: Chen Minqiang <ptpt52@gmail.com>
---
 target/linux/ath79/image/lzma-loader/src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/linux/ath79/image/lzma-loader/src/Makefile b/target/linux/ath79/image/lzma-loader/src/Makefile
index c7bb935a030c..f5823c5a654c 100644
--- a/target/linux/ath79/image/lzma-loader/src/Makefile
+++ b/target/linux/ath79/image/lzma-loader/src/Makefile
@@ -31,7 +31,7 @@ OBJDUMP		:= $(CROSS_COMPILE)objdump
 BIN_FLAGS	:= -O binary -R .reginfo -R .note -R .comment -R .mdebug \
 		   -R .MIPS.abiflags -S
 
-CFLAGS		= -D__KERNEL__ -Wall -Wstrict-prototypes -Wno-trigraphs -Os \
+CFLAGS		= -D__KERNEL__ -Wall -Wstrict-prototypes -Wno-trigraphs -O2 \
 		  -fno-strict-aliasing -fno-common -fomit-frame-pointer -G 0 \
 		  -mno-abicalls -fno-pic -ffunction-sections -pipe -mlong-calls \
 		  -fno-common -ffreestanding -fhonour-copts -nostartfiles \
